### DISCUSSION TITLE
Wrapping items/keys/values

### DISCUSSION THREAD

**Discussion by adriangb** [pydantic/pydantic#5654] [Question]
It would be really interesting if ValidatorFunctionWrapHandler was a real class and offered specialized methods for validating items/keys/values.

```python
class FunctionWrapValidator:
  def __call__(…): …

class IterableWrapValidator(FunctionWrapValidator):
  def validate_item(self, item: Any) -> Any: …
```

Currently you can wrap a list validator w/ a function validator but you can only choose to validate the entire list or not. If we could wrap “each item” that would unlock a lot of cool possibilities and make things like https://github.com/pydantic/pydantic/issues/2274#issuecomment-1528052185 a lot easier (by getting rid of the marker bit) and maybe custom collection validators that don’t need an intermediary dict/list validator.

Maybe model validators could also offer `validate_field` so that you can dispatch a single field at a time.

**Comment by bkis** [comment]
> Maybe model validators could also offer `validate_field` so that you can dispatch a single field at a time.

This would be great!
